src/lib/ndpi_main.c

```diff
@@ -6922,3 +6940,19 @@ int ndpi_check_dga_name(struct ndpi_detection_module_struct *ndpi_str,
 
   return(rc);
 }
+
+/* ******************************************************************** */
+
+ndpi_protocol_extid_t ndpi_intid_to_extid(u_int16_t protoId) {
+  if(protoId >= NDPI_LAST_IMPLEMENTED_PROTOCOL)
+    return NDPI_PROTOCOL_EXT_UNKNOWN;
+
+  return(ndpi_ext_protoid[protoId]);
+}
+
+u_int16_t ndpi_extid_to_intid(ndpi_protocol_extid_t extId) {
+  if(extId >= NDPI_PROTOCOL_EXT_MAX)
```

Comment:
As `exitId` is most likely an `int` (depending on the highest value in `ndpi_protocol_extid_t` enum) a check for `extId < 0` seems reasonable to me.

Reply:
This check is to avoid reading beyond the bounds of ndpi_int_protoid at line 6957.

Reply:
I know. I meant that `extId` is an int and if you check for `NDPI_PROTOCOL_EXT_MAX`, you should also check for `< 0`. That way you can make sure that all values for `extId` are valid indices.